### PR TITLE
add profile

### LINKED ISSUE #1123: feature: Does apisix support distinguish configuration of different environments by environment variables ?

### PR DESCRIPTION
**Pull request by qiujiayu**
Fix #1123

### CODE REVIEW

**Comment by moonming** [comment]
@qiujiayu please take a look at CI, thx

File: lua/apisix/core/profile.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local _M = {
+    version = 0.1,
+    profile = os.getenv("APISIX_PROFILE"),
+    apisix_home = ngx and ngx.config.prefix()
```
Comment (membphis) [inline lua/apisix/core/profile.lua]: Do we need to set a default path if the `profile.lua` was not running in OpenResty ENV.

Comment (qiujiayu) [reply lua/apisix/core/profile.lua]: Ok, I have been set the value `(ngx and ngx.config.prefix()) or ""` for the var apisix_home

**Comment by membphis** [comment]
Looks good to me, we can merge it soon.

**Comment by membphis** [comment]
@qiujiayu Thank you very much, it has been merged.

**Pull request merged**